jschon/vocabulary/__init__.py

```diff
@@ -105,6 +121,17 @@ def __str__(self) -> str:
 KeywordClass = Type[Keyword]
 
 
+class AnnotationMixin(Keyword):
```

Comment:
I'm going to suggest dropping `AnnotationMixin` and just going with `AnnotationKeyword(Keyword)`, with the same class definition as it had within the annotation.py module. There are two aspects to this:
1. A mixin class is usually not used as a primary ancestor class; it doesn't form part of the class hierarchy of whatever it's being mixed into (and I think trying call a mixin method via `super()` usually doesn't make sense); it adds rather than modifies behaviour.
2. In light of point 1, a better name for `AnnotationMixin` in this context might be something like `AbstractAnnotationKeyword` or `BaseAnnotationKeyword`, but I don't think this level of abstraction is needed (until such time as there is a use case for it).

Reply:
I'm happy to do this, although I have some thoughts and questions.  I'm also happy to implement what you've said for this PR and move this discussion out to an issue if you think there's more discussion worth having.

I definitely agree that `AbstractAnnotationKeyword` would be a better name.  It occurred to me that implementing the method made it different from `ApplicatorMixin`** but I didn't think of anything to do about it at the time.

I made `AnnotationMixin` with an eye towards the API for extension keywords, which might both annotate and do something else.  And actually, I forgot that there is already a use case for it: `format`, which (depending on the vocabulary and configuration) can be both an annotation and assertion (I try t forget that `format` exists at all, or I would have refactored `FormatKeyword` to inherit from it.  That would be the correct usage to demonstrate for custom keyword authors, and if you want to keep `AbstractAnnotationKeyword` I would do that refactoring.

**if I may be pedantic 😅 , I'd actually suggest renaming `ApplicatorMixin` and friends `SchemaLocationMixin`, etc.  `$defs` uses this hierarchy but is not an applicator, while `$ref` and `$dynamicRef` which are applicators, do not, which I found confusing.  In draft-next (and ambiguously in earlier drafts), [`contentSchema` ought to inherit from this mixin as well](https://github.com/json-schema-org/json-schema-spec/pull/1352).  If you'd be receptive to such a change, I can file a separate issue and make a PR.

Reply:
Could you please create an issue for renaming the applicator mixin classes? I agree in principle, though I'm thinking of renaming as follows (but I'm open to discussion on this):

ApplicatorMixin -> SubschemaMixin
Applicator -> Subschema
ArrayApplicator -> ArrayOfSubschemas
PropertyApplicator -> ObjectOfSubschemas

Then, class definitions would look like:

```py
class IfKeyword(Keyword, Subschema):
    ...

class AnyOfKeyword(Keyword, ArrayOfSubschemas):
    ...

class PropertiesKeyword(Keyword, ObjectOfSubschemas):
    ...
```

And please create a separate issue and PR for the `contentSchema` change.

Thanks!

Reply:
Will do - this is also related to the stuff I alluded to regarding `"propertyDependencies"` (which is an object of object of schemas instead of just an object of schemas), so I might take a day or two while I make sure I'm covering everything.  If you can take a look at PRs 57, 65, 66, and 67 (which are all quite short and simple) in the meantime that will simplify my life immensely due to git's awful submodule behavior (why do submodules not automatically update to match the main repo's branch state?)

Reply:
>  If you can take a look at PRs 57, 65, 66, and 67

Certainly - I should be able to get through them tomorrow.

> git's awful submodule behavior

I don't understand it either :shrug:

Reply:
OK, I've merged 57. I'd like to spend a bit more time on 65 and 66, however, to fully understand the issues they resolve and the implications of the changes. I'll only have time for this on the weekend, unfortunately.

Reply:
@marksparkza no problem!  Both of those PRs address JSON Schema Test Suite failures - I'll update the descriptions with the details.